### PR TITLE
Do not run placement service as root

### PR DESCRIPTION
**Pull request by gibizer**
Depends-On: https://github.com/openstack-k8s-operators/tcib/pull/102
Implements: https://issues.redhat.com/browse/OSPRH-1374
Depends-On: https://github.com/openstack-k8s-operators/openstack-operator/pull/591 (to pick up a new tcib image)

### CODE REVIEW

**Comment by gibizer** [comment]
/test precommit-check
```
   * could not run steps: step precommit-check failed: test "precommit-check" failed: could not watch pod: the pod ci-op-gids8xyf/precommit-check failed after 2m3s (failed containers: test): ContainerFailed one or more containers exited \
```

File: api/bases/placement.openstack.org_placementapis.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.11.1
```
Comment (SeanMooney) [inline api/bases/placement.openstack.org_placementapis.yaml]: this is proably not intentional

**Review by SeanMooney — CHANGES_REQUESTED** [review]

File: api/bases/placement.openstack.org_placementapis.yaml (hunk 1 above)
Comment (gibizer) [reply api/bases/placement.openstack.org_placementapis.yaml]: true. fixed.

**Review by bogdando — COMMENTED** [review]
lgtm

**Comment by gibizer** [comment]
/hold

This should not work as kolla_set_config should fail.

Also we cannot simply switch to the placement user in the container as that user is not in the kolla group:
```
❯ oc rsh placement-7ccc6bc7f5-cfchj
Defaulted container "placement-api" out of: placement-api, init (init)
sh-5.1# id
uid=0(root) gid=0(root) groups=0(root)
sh-5.1# cat /etc/passwd | grep placement
placement:x:997:994:OpenStack Placement:/:/bin/bash
sh-5.1# cat /etc/group | grep kolla
kolla:x:42400:
sh-5.1# 
```

**Comment by gibizer** [comment]
And we have a strange dbsync command:
```
	DBSyncCommand = "/usr/local/bin/kolla_set_configs && su -s /bin/sh -c \"placement-manage db sync\" placement"
```

**Comment by gibizer** [comment]
We need a fix in the tcib image first https://github.com/openstack-k8s-operators/tcib/pull/102

**Comment by gibizer** [comment]
tcib fix merged but there is still no container image built in https://quay.io/repository/podified-antelope-centos9/openstack-placement-api?tab=tags&tag=latest so we wait...

**Comment by gibizer** [comment]
container promotion is blocked on https://issues.redhat.com/browse/OSPCIX-119

**Review by SeanMooney — APPROVED** [review]
https://issues.redhat.com/browse/OSPCIX-119 is closed 
and there was a successful promotion yesterday
https://review.rdoproject.org/zuul/build/97b34cff01744289b42f01944b777616
so i belive we can now proceed with this

@gibi if you agree then feel free to drop your hold

**Comment by gibizer** [comment]
This has the same limitation as nova-operator has. We cannot remove the usage of root from the container as kolla_set_config needs sudo.

**Comment by gibizer** [comment]
/hold
We need to wait for https://github.com/openstack-k8s-operators/openstack-operator/pull/591 to pick up the new dataplane-operator that will pick up the new placement-api tcib image.

**Comment by gibizer** [comment]
> /hold We need to wait for [openstack-k8s-operators/openstack-operator#591](https://github.com/openstack-k8s-operators/openstack-operator/pull/591) to pick up the new dataplane-operator that will pick up the new placement-api tcib image.

I mixed up this needs a new placement-operator to pick up a new placement-api tcib image. But we are in placement-operator so we don't have to wait for that.

**Review by bogdando — APPROVED** [review]

**Pull request merged**